charts/astronomer/templates/_helpers.yaml

```diff
@@ -171,10 +171,19 @@ azure:
 {{- end }}
 
 {{- define "houston_environment" }}
+{{- /* Dynamically created envs */ -}}
 {{- range $i, $config := .Values.houston.env }}
 - name: {{ $config.name }}
   value: {{ $config.value | quote }}
-  {{- end }}
+{{- end }}
+{{- /* Dynamically created secret envs */ -}}
+{{- range $i, $config := .Values.houston.secret }}
+- name: {{ $config.envName }}
+  valueFrom:
+    secretKeyRef:
+      name: {{ $config.secretName }}
+      key: {{ default "value" $config.secretKey }}
+{{- end }}
```

Comment:
only concern, this block will be in all houston related pods e.g.: upgrade-deployments/cleanup etc.
wdyt to extract to new `{{ -define ` block?

Reply:
I think that might be desirable, since they also pull configuration from the same Houston `config` module. I look at these new jobs as being _exactly_ the same Houston proper, just doing handling specialized things in the backgroud.